### ISSUE TITLE
Photoshop fails to start up

### ISSUE THREAD

**Issue by fridenity0777** [YoungFellow-le/photoshop-22-linux#13]
### A clear description of the bug:
Photoshop failed to start up; Instead, launching it outputs a very long error message.

It used to be working a few weeks ago, and I assume this is more of a wine issue. But I've searched the error up and no one seemed to have the same problem as I did, so I thought I'd bring it up just in case someone else is also experiencing this.

### The output from the installer:
After running:
```bash
bash -c '$INSTALL_DIR/photoshop-22-linux/scripts/launcher.sh'
```
[**Here**](https://github.com/YoungFellow-le/photoshop-22-linux/files/9995253/err.log) is the command output

### System info:
 - **Linux distribution:** Artix Linux, kernel `6.0.7-artix1-1`
 - **Desktop environment:** None (I use a WM)
 - **Wine version:** `7.20`

**Comment by YoungFellow-le** [comment]
First of all, **thank you** for filling the bug form out.

Secondly, the issue seems to be with your `wine` installation.
All those `.dll`s that are not found are system libraries....
However, I'm not sure what to recommend.
Maybe try reinstalling `wine`?
Let me know how it goes.

**Comment by Fridenity** [comment]
I have reinstalled `wine`, deleted the entire cloned repo and redone everything. The same problem persists.
[Here](https://github.com/YoungFellow-le/photoshop-22-linux/files/9998083/out.log) is the installation output. Some of the output at the top is cut off, but I think the relevant info (wine errors) are all in there.

**Comment by eocow** [comment]
I have a similar issue when I used wine 7.20 where nothing would launch after install.

**Comment by YoungFellow-le** [comment]
Have you guys seen this:
https://www.linuxquestions.org/questions/linux-software-2/wine-cant-find-the-dlls-193935/

**Comment by YoungFellow-le** [comment]
Just updated to wine v7.20.
Will test now.

**Comment by YoungFellow-le** [comment]
To enumerate this further:
1. Do you get the same error when running photoshop.exe using the default wine prefix?
2. Do you get similar errors when running other applications?

**Comment by Fridenity** [comment]
1. Yes, the same error.
2. I've noticed that in past applications I've installed via bottles output the same error when using the default wine prefix, but they work perfectly fine in their own respective bottle's wine prefix.
3. I have now installed those two packages you mentioned. I have reinstalled photoshop all over again, and the issue persists. 
I have also tried other wine implementations (`wine-staging`, `wine-ge-custom`) with no difference in results.

I'm not familiar with wine at all. Sorry if I've done something trivially incorrect.
If you want to take this conversation elsewhere (so as not to clutter up the GitHub issue comments), add me on discord `Fridella#0777` and continue there.

**Comment by eocow** [comment]
I just tried it in the 7.21 devel branch on wine and it seems to work perfectly so far. will edit if i run into any issues

**Comment by YoungFellow-le** [comment]
> I'm not familiar with wine at all. Sorry if I've done something trivially incorrect.

No need to apologise :)

I'd rather you keep this issue here, in case someone else faces the same issue.
Do keep me updated.

**Comment by eocow** [comment]
I have been using it for a day now and have ran into a few issues:

1. failure to export files as seen in this image 
![image](https://user-images.githubusercontent.com/70074743/203230424-3752ed9c-fd85-49ce-8757-40ae2253d0e5.png)

2. Raw images work but the window that pops up allowing you to edit the raw image will timeout for some reason and will not allow input until you alt-tab out and in

**Comment by CHJ85** [comment]
Same here. I'm using Wine version 6.0.3 as it's the latest one that'll work on Ubuntu.
I tried the PPA which comes with version 7, but not as wine package, but rather wine-stable, which this script doesn't recognize for some reason.
How do I fix this issue?

**Comment by CHJ85** [comment]
Does **not** work on Wine 7.0.1.
Does **not** work on 7.22-staging.
I think you should just bundle a working version of Wine in with the Photoshop scripts instead of relying on the system-wide installation.

**Comment by Fridenity** [comment]
> Does **not** work on Wine 7.0.1.
> Does **not** work on 7.22-staging.
> I think you should just bundle a working version of Wine in with the Photoshop scripts instead of relying on the system-wide installation.

Yeah, I agree as well. I still don't know why the wine/wine staging, packaged on my artix repo doesn't work (and I don't know which wine would've worked either yet). I've even tried deleting my whole original wine prefix, nothing seems to be working.

**Comment by Fridenity** [comment]
**Update**: I tried reinstalling this again, but there are new errors.
> [Installation output](https://github.com/YoungFellow-le/photoshop-22-linux/files/10189585/err.new.log)
> **Command output**:
```sh
$ bash -c '/home/.../Applications/photoshop/photoshop-22-linux/scripts/launcher.sh'
/home/.../Applications/photoshop/photoshop-22-linux/scripts/launcher.sh: line 2: cd: /home/.../Applications/photoshop/photoshop-22-linux/Ps-prefix/drive_c/Program Files/Adobe/Adobe Photoshop 2021/: No such file or directory
002c:fixme:winediag:LdrInitializeThunk wine-staging 7.22 is a testing version containing experimental patches.
002c:fixme:winediag:LdrInitializeThunk Please mention your exact version when filing bug reports on winehq.org.
007c:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
007c:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
007c:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
007c:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
007c:fixme:wineusb:query_id Unhandled ID query type 0x5.
007c:fixme:wineusb:query_id Unhandled ID query type 0x5.
007c:fixme:wineusb:query_id Unhandled ID query type 0x5.
007c:fixme:wineusb:query_id Unhandled ID query type 0x5.
WARNING: radv is not a conformant Vulkan implementation, testing use only.
Application could not be started, or no application associated with the specified file.
ShellExecuteEx failed: File not found.
```
TL;DR: Glancing through the installation output, there seems to be an error with `ShellExecuteEx` and installing `vk3d`
This also seems to be related to #15 as there are errors about the google drive link being bad.

**Comment by Fridenity** [comment]
**Update**: I have spent some time debugging and pinpointed a few bugs.
1. Less of a bug, and more of a heads-up. Consider putting this in the README:
* [This](https://github.com/YoungFellow-le/photoshop-22-linux/blob/main/scripts/installer.sh#L108) and [this gdown calls](https://github.com/YoungFellow-le/photoshop-22-linux/blob/main/scripts/installer.sh#L121) is causing issues on some occasions. Running this command in a separate shell outputs the following:
```bash
$ gdown "1esUAZkejzJARub9cessbVeUCDlzzzcQG" -O installation_files/ps_components.tar.xz

Access denied with the following error:

 	Cannot retrieve the public link of the file. You may need to change
	the permission to 'Anyone with the link', or have had many accesses.

You may still be able to access the file from the browser:

	 https://drive.google.com/uc?id=1esUAZkejzJARub9cessbVeUCDlzzzcQG
```
* **Uninstalling `gdown` by `pip uninstall gdown`, and then reinstalling with `pip install gdown` fixes this issue.** If not, here are some other potential ways to fix the issue: [[1]](https://github.com/wkentaro/gdown/issues/43#issuecomment-638232081) [[2]](https://github.com/wkentaro/gdown/issues/43#issuecomment-1278345755) [[3]](https://github.com/wkentaro/gdown/issues/43#issuecomment-1328098120)
* Opening this link in the browser and downloading the files manually in the relevant directory is also viable as a workaround. 
* Interestingly enough, why is the `gdown` gdown link from the two command instances different? (`1esUAZkejzJARub9cessbVeUCDlzzzcQG` & `1VqIUUzCDuyxOXM-q99ySKEKU_n8DbB26 `), considering this links to the same file anyways.
2. [This sh call](https://github.com/YoungFellow-le/photoshop-22-linux/blob/main/scripts/installer.sh#L214) is intended to work only if the script is strictly POSIX, and while most Linux systems have `/bin/sh` as `bash`, some major distributions such as Ubuntu and Fedora are now using `dash` as `/bin/sh`, which is a much stricter shell also with POSIX compliance.
* **You probably meant to type `./scripts/setup_vkd3d_proton.sh install`, because that vkd3d setup script is already shebang'd with [`#!/usr/bin/env bash`](https://github.com/YoungFellow-le/photoshop-22-linux/blob/main/scripts/setup_vkd3d_proton.sh#L1)**. So running the script as an executable will just use bash, and not `sh` (which is what is happening instead)
3. `winetricks` is still missing some DLL files.
* Running `bash -c '$INSTALL_DIR/photoshop-22-linux/scripts/launcher.sh'` exits with the code 53 and outputs the following:
```bash
002c:fixme:winediag:LdrInitializeThunk wine-staging 7.22 is a testing version containing experimental patches.
002c:fixme:winediag:LdrInitializeThunk Please mention your exact version when filing bug reports on winehq.org.
007c:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
007c:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
007c:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
007c:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
007c:fixme:wineusb:query_id Unhandled ID query type 0x5.
007c:fixme:wineusb:query_id Unhandled ID query type 0x5.
007c:fixme:wineusb:query_id Unhandled ID query type 0x5.
007c:fixme:wineusb:query_id Unhandled ID query type 0x5.
010c:err:module:import_dll Library MSVCP140_CODECVT_IDS.dll (which is needed by L"C:\\Program Files\\Adobe\\Adobe Photoshop 2021\\AdobePDFSettings.dll") not found
010c:err:module:import_dll Library AdobePDFSettings.dll (which is needed by L"C:\\Program Files\\Adobe\\Adobe Photoshop 2021\\photoshop.exe") not found
010c:err:module:LdrInitializeThunk Importing dlls for L"C:\\Program Files\\Adobe\\Adobe Photoshop 2021\\photoshop.exe" failed, status c0000135
```
* I have tried to install some additional DLLs with `winetricks`, notably `vcrun2019 vcrun2012 vcrun2013 vcrun2010` [(Referenced)](https://github.com/hitman249/wine-launcher/issues/40). The problem still persists.
* I have also tried reinstalling wine, switching to a different development branch (`wine-staging`), using a fork (`wine-ge-custom`), deleting `$XDG_DATA_HOME/wine`, nothing have worked.

**Comment by YoungFellow-le** [comment]
Thank you for the these pointers @Fridenity.

> * Interestingly enough, why is the `gdown` gdown link from the two command instances different? (`1esUAZkejzJARub9cessbVeUCDlzzzcQG` & `1VqIUUzCDuyxOXM-q99ySKEKU_n8DbB26 `), considering this links to the same file anyways.

I have one character to reply to this:
:sweat_smile:



> 2\. [This sh call](https://github.com/YoungFellow-le/photoshop-22-linux/blob/main/scripts/installer.sh#L214) is intended to work only if the script is strictly POSIX, and while most Linux systems have `/bin/sh` as `bash`, some major distributions such as Ubuntu and Fedora are now using `dash` as `/bin/sh`, which is a much stricter shell also with POSIX compliance.

I literally copy pasted that from the script I originally forked mine from.
I implemented both suggestions in commits da6ba4b06bfc56ab07b348b889fdd3c2d207ceec and 525b4021fcf22cffdb955b83a6cac6395a4ec1aa.

As for the `gdown` error, can you add it to the README?
You can create a PR, thanks.

**Comment by fridenity0777** [comment]
I'm sorry but why was this issue closed? It's not been resolved yet. If you closed because of inactivity, I'd appreciate it if you let others who might have the same problem know about it.

**Comment by fridenity0777** [comment]
Sorry for the last comment that I deleted, I tried reinstalling photoshop with the latest commit again today and photoshop finally launches, so thank you for your hard work!

**Comment by YoungFellow-le** [comment]
No problem, I just didn't hear from you for a while.
